AssetRipper.GUI/ConsoleOptions.cs

```diff
@@ -0,0 +1,350 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+using System.CommandLine;
+using System.CommandLine.Invocation;
+
+using AssetRipper.Core.IO.MultiFile;
+using AssetRipper.Core.Utils;
+using AssetRipper.Library;
+
+
+namespace AssetRipper.GUI
+{
+	internal enum ExitCode
+	{
+		ExtractingError = 1,
+		ArgumentError = 2,
+	}
+
+	internal static class ConsoleOptions
+	{
+		public const string DefaultLogFileName = "AssetRipper.log";
+		public const string DefaultOutputPath = "Ripped";
+		public const bool DefaultQuit = false;
+		public const bool DefaultVerbose = false;
+
+		private static InvocationContext? _context;
+
+		#region Utilities
+
+		/// <summary>
+		/// Converts the name of the given type to hyphen-case, with a leading hyphen.
+		/// For example: ConsoleOptions -> -console-options (if using the default separator)
+		/// </summary>
+		/// <param name="input">The type to get the name of</param>
+		/// <param name="separator">Char to use as separator</param>
+		/// <returns></returns>
+		public static string OptionNameFromType(Type input, char separator = '-')
+		{
+			IEnumerable<string> result = input.Name.Select((character) =>
+			{
+				if (char.IsUpper(character))
+					return $"{separator}{char.ToLower(character)}";
+
+				return character.ToString();
+			});
+
+			return $"{separator}{string.Join("", result)}";
```

Comment:
Use a `StringBuilder` to allocate one string instead of dozens.

Reply:
@DecentM todo